### PR TITLE
feat(sitemap): allow customizing the output name

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

See https://docusaurus.canny.io/feature-requests/p/support-multi-instance-sitemaps-plugins

Since sitemap files can be manually uploaded/checked-in, the name is mostly irrelevant.

## Test Plan

We don't have tests for this...

### Test links

Deploy preview: https://deploy-preview-7469--docusaurus-2.netlify.app/docs/api/plugins/@docusaurus/plugin-sitemap/

### CODE REVIEW

File: website/docs/api/plugins/plugin-sitemap.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ Accepted fields:
 | `changefreq` | `string` | `'weekly'` | See [sitemap docs](https://www.sitemaps.org/protocol.html#xmlTagDefinitions) |
 | `priority` | `number` | `0.5` | See [sitemap docs](https://www.sitemaps.org/protocol.html#xmlTagDefinitions) |
 | `ignorePatterns` | `string[]` | `[]` | A list of glob patterns; matching route paths will be filtered from the sitemap. Note that you may need to include the base URL in here. |
+| `filename` | `string` | `sitemap.xml` | The path to the created sitemap file, relative to the output directory. |
```
Comment (slorber) [inline website/docs/api/plugins/plugin-sitemap.md:43]: also add to example configuration below?

**Review by slorber — APPROVED** [review]
🤷‍♂️ ok to add this, but I'm not sure to understand the use-case in practice

> The docs state that "all Docusaurus content plugins can support multiple plugin instances". This support should extent to other plugins such as plugin-sitemap. For example, we want to weight docs differently in SEO based on their doc ID.

Yes but... all the sitemap plugins would receive the same routes, so unless you plan with the `ignorePatterns ` you end up with the same result 🤷‍♂️ 

Now, about "weighting" some docIds, I don't know what the op means exactly

https://developers.google.com/search/docs/advanced/sitemaps/build-sitemap#general-guidelines

> Google ignores <priority> and <changefreq> values.

**Comment by dsmmcken** [comment]
> Yes but... all the sitemap plugins would receive the same routes, so unless you plan with the ignorePatterns you end up with the same result

Thanks for this. I actually have a use case.

I have a section of my site behind http-auth, and would like to exclude those pages from my sitemap that gets submitted to google for indexing. Google continually warns me it can't reach all those pages in the sitemap.

But I also need a full sitemap to pass to the algolia self hosted crawler to still index those pages (that crawls with http-auth).

I intended to publish two sitemaps, one for algolia indexing, one for google, this will make that easier.

**Comment by slorber** [comment]
Thanks for clarifying, I wouldn't have been able to think of this use-case 🤪

**Pull request merged**